src/Voyager.php

```diff
@@ -68,12 +68,24 @@ public static function can($permission)
 
         if ($exist) {
             $user = User::find(Auth::id());
-            if ($user == null) {
+            if ($user == null || !$user->hasPermission($permission)) {
                 throw new UnauthorizedHttpException(null);
             }
-            if (!$user->hasPermission($permission)) {
-                throw new UnauthorizedHttpException(null);
+        }
+    }
+
+    public static function have($permission)
+    {
+        // Check if permission exist
+        $exist = Permission::where('key', $permission)->first();
+
+        if ($exist) {
+            $user = User::find(Auth::id());
```

Comment:
This could become `$user =  auth()->user();`.
Then it will no longer require a SQL query.

The the user could eager-load their permissions, that will help limit the amount of queries run when checking for multiple permissions.
```
if (! $user->relationLoaded('permissions')) {
    $user->load('permissions');
}
```

Then the last step is to remove the part that checks if the permission exists and replace it with a method to holds all the permissions. We could for this use the `Voyager` facade, then the Voyager instance could hold all existing permissions. Maybe something like this:
```
// Check if permission exists
$exists = \TCG\Voyager\Facades\Voyager::permissionExists('visit_admin');

// Method in Voyager instance
protected $permissions;
public function permissionExists($key)
{
    if (is_null($this->permissions)) {
        $this->permissions = Permission::all();
    }

    return $this->permissions->where('key', $key)->count() > 0;
}
```

> I now that this code have been here for a while, and is not your fault 👍